README.rst

```diff
@@ -0,0 +1,793 @@
+  may be used to reference URLs in separate attributes outside of a `purl` or in
+  a `purl` qualifier.
+
+- Version control system (VCS) URLs such `git://`, `svn://`, `hg://` or as
+  defined in Python pip or SPDX download locations are NOT valid `purl` types.
+  They are a closely related, compact and uniform way to reference vcs URLs.
+  They may be used as references in separate attributes outside of a `purl` or
+  in a `purl` qualifier.
+
+
+Rules for each `purl` component
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A `purl` string is an ASCII URL string composed of six components.
+
+Some components are allowed to use other characters beyond ASCII: these
+components must then be UTF-8-encoded strings and percent-encoded as defined in
+the "Character encoding" section.
+
+The rules for each component are:
+
+- **type**:
+
+  - The package `type` is composed only of ASCII letters and numbers, '.', '+'
+    and '-' (period, plus, and dash)
+  - The `type` cannot start with a number
+  - The `type` cannot contains spaces
+  - The `type` must NOT be percent-encoded
+  - The `type` is case insensitive. The canonical form is lowercase
+  - Since a `purl` never contains a URL Authority, its `type` must not be
+    suffixed with double slash as in 'docker://' and should use instead
+    'docker:'. Otherwise this would be an invalid URI per rfc3986 at
+    https://tools.ietf.org/html/rfc3986#section-3.3::
+
+        If a URI does not contain an authority component, then the path
+        cannot begin with two slash characters ("//").
+
+    While it is acceptable to use such '://' suffix, its is not significant and
+    not needed for unambiguous parsing even if it make a `purl` look likea a
+    familiar web URL. In its canonical form, a `purl` must NOT use such '://'
+    `type` suffix. 
+  - `purl` parsers must accept URLs with such '://' and must ignore the '//'.
+  - `purl` builders must not create invalid URLs with such double slash '//'.
+  - The `type` is followed by a ':' separator
+  - For example these two purls are strictly equivalent and the first is in
+    canonical form. The second `purl` with a '//' is an acceptable `purl` but is
+    an invalid URI/URL per rfc3986::
+
+            gem:ruby-advisory-db-check@0.12.4
+            gem://ruby-advisory-db-check@0.12.4
+
+
+- **namespace**:
+
+  - The optional `namespace` contains zero or more segments, separated by slash
+    '/'
+  - Leading and trailing slashes '/' are not significant and should be stripped
+    in the canonical form. They are not part of the `namespace`
+  - Each `namespace` segment must be a percent-encoded string
+  - When percent-decoded, a segment:
+
+    - must not contain a '/'
+    - must not be empty
+
+  - A URL host or Authority must NOT be used as a `namespace`. Use instead a
+    `repository_url` qualifier. Note however that for some types, the
+    `namespace` may look like a host.
+
```

Comment:
Again, I'd like to understand why a namespace cannot use URL host?

Reply:
It makes sense for docker, so that would be type specific. But in the general case, we are avoiding  the reuse of the authority part of a URL. I updated things to support docker namespace for this